packages/headless/src/controllers/insight/attach-to-case/headless-attach-to-case.test.ts

```diff
@@ -0,0 +1,121 @@
+import {attachedResults, configuration} from '../../../app/reducers';
+import {
+  attachResult,
+  detachResult,
+} from '../../../features/attached-results/attached-results-actions';
+import {InsightAppState} from '../../../state/insight-app-state';
+import {createMockAttachedResult} from '../../../test/mock-attached-results';
+import {
+  buildMockInsightEngine,
+  MockInsightEngine,
+} from '../../../test/mock-engine';
+import {buildMockInsightState} from '../../../test/mock-insight-state';
+import {buildMockRaw} from '../../../test/mock-raw';
+import {buildMockResult} from '../../../test/mock-result';
+import {
+  AttachToCase,
+  buildAttachToCase,
+  SearchResult,
+} from './headless-attach-to-case';
+
+describe('insight attach to case', () => {
+  let engine: MockInsightEngine;
+  let state: InsightAppState;
+  let attachToCase: AttachToCase;
+
+  function initController() {
+    engine = buildMockInsightEngine();
+    engine.state = state;
+    attachToCase = buildAttachToCase(engine);
+  }
+
+  beforeEach(() => {
+    state = buildMockInsightState();
+    initController();
+  });
+
+  it('it adds the correct reducers to the engine', () => {
+    expect(engine.addReducers).toHaveBeenCalledWith({
+      configuration,
+      attachedResults,
+    });
+  });
+
+  it('calling #isAttached should return false if there are no results attached.', () => {
+    const mockSearchResult: SearchResult = {
+      title: '',
+      raw: {
+        permanentid: 'foo',
+        urihash: undefined,
+      },
+    };
+    expect(attachToCase.isAttached(mockSearchResult)).toBeFalsy();
+  });
+
+  it('calling #isAttached should return true if the result is attached with the permanentId.', () => {
+    const testPermanentId = 'test_permanentid';
+    const mockAttachedResult = createMockAttachedResult({
+      permanentId: testPermanentId,
+      uriHash: undefined,
+    });
+    engine.state.attachedResults.results.push(mockAttachedResult);
+
+    const mockSearchResult = buildMockResult({
+      raw: buildMockRaw({
+        permanentid: testPermanentId,
+      }),
+    });
+    expect(attachToCase.isAttached(mockSearchResult)).toBeTruthy();
```

Comment:
toBe(true) or toBeTruthy(), that is the question...

more seriously though given it's a boolean I would recommend `toBe(true)` and `toBe(false)` as it's more specific than truthy/falsy.

Reply:
Sure I was not aware of any preferences and vscode suggested `toBeTruthy` first haha